engine/apps/mobile_app/models.py

```diff
@@ -16,6 +17,19 @@ def get_expire_date():
     return timezone.now() + timezone.timedelta(seconds=MOBILE_APP_AUTH_VERIFICATION_TOKEN_TIMEOUT_SECONDS)
 
 
+class ActiveFCMDeviceQuerySet(models.QuerySet):
+    def filter(self, *args, **kwargs):
+        return super().filter(*args, **kwargs, active=True)
+
+
+class FCMDevice(BaseFCMDevice):
+    active_objects = ActiveFCMDeviceQuerySet.as_manager()
+
+    @classmethod
+    def get_active_device_for_user(cls, user: User) -> "FCMDevice" | None:
+        return cls.active_objects.filter(user=user).first()
```

Comment:
I'm not in the context of FCM, but shouldn't we consider the last connected device as active?

Reply:
There is explicitly an `active` column for this (the setting of this to true/false is all managed by the `fcm_django` library).

**More context**
We have `ONE_DEVICE_PER_USER` set to `True` ([here](https://github.com/grafana/oncall/blob/dev/engine/settings/base.py#L621)) which according to the docs:
> true if you want to have only one active device per registered user at a time

There was an edge case with this setting though that I'm trying to solve for here. If the user had one active and one inactive device in `fcm_django_fcmdevice` table, using `.first()` could possibly return the inactive device

Reply:
if we are sure there is the only one device, can we use `.get()`?

Reply:
good point 👍 updated this and added some tests around this logic. also verified our databases that this is currently the case.